general/annotation.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_ANNOTATION_HPP
+#define MFEM_ANNOTATION_HPP
+
+#include "../config/config.hpp"
+
+#ifdef MFEM_USE_CALIPER
+
+#include <caliper/cali.h>
+#include <caliper/cali-manager.h>
+#define MFEM_MARK_FUNCTION CALI_CXX_MARK_FUNCTION
+#define MFEM_MARK_REGION_BEGIN(s) CALI_MARK_BEGIN(s)
+#define MFEM_MARK_REGION_END(s) CALI_MARK_END(s)
+
+#else
+
+#define MFEM_MARK_FUNCTION
+#define MFEM_MARK_REGION_BEGIN(s)
+#define MFEM_MARK_REGION_END(s)
```

Comment:
I maintain that my preference goes for something like `MFEM_BENCHMARK_FUNCTION`. I would also keep it shorter with `MFEM_BENCHMARK_BEGIN` and `MFEM_BENCHMARK_END`.

Reply:
My preference is for `MFEM_PERF_FUNCTION`, `MFEM_PERF_BEGIN` and `MFEM_PERF_END`

Reply:
Set to:
MFEM_PERF_FUNCTION
MFEM_PERF_BEGIN(s)
MFEM_PERF_END(s)